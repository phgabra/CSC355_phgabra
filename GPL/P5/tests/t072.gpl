string s1 = 40;

string s2 = 2;

// string concatinaton: "40" + "2" = "402"
string s3 = s1 + s2; 

// the right hand sides should be evaluated and then the result cast to a string
string s4 = 38.86 + 3.14;

string s5 = 41.12 + .88;

string s6 = 2 * 21;

string s7 = 84 / 2.0;

string s8 = 2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2+2;

string s9 = 84-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2-2;
