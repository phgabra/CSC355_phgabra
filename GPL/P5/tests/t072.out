gpl.cpp::main()
  input file(tests/t072.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(false)
  symbol_table(true)
  graphics(false)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().

No errors found (parser probably worked correctly).


Printing the symbol table from main()
string s1 = "40"
string s2 = "2"
string s3 = "402"
string s4 = "42.000000"
string s5 = "42.000000"
string s6 = "42"
string s7 = "42.000000"
string s8 = "42"
string s9 = "42"
Graphics is turned off by the Makefile.  Program exiting.

gpl.cpp::main() done.
